pkg/clusterconfig/config.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2019, Oath Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package clusterconfig
```

Comment:
Since we only need the token fetch / rotation logic for the on prem version, can we only port the necessary codebase associated with it? I would rather not bring in the entire clusterconfig here as it is not really needed for the functionality of this feature.